subsys/net/ip/net_private.h

```diff
@@ -108,6 +108,13 @@ static inline u16_t net_calc_chksum_icmpv4(struct net_pkt *pkt)
 }
 
 static inline u16_t net_calc_chksum_udp(struct net_pkt *pkt)
+{
+	u16_t chksum = net_calc_chksum(pkt, IPPROTO_UDP);
+
+	return chksum == 0U ? 0xffff : chksum;
```

Comment:
Like, 0 is unsigned, but 0xffff - not?

Reply:
See commit 24d71431e9c9da493ce00eb284ac9b0b2cb1bdca